meteor-slingshot/services/cloudflare-r2.js

```diff
@@ -0,0 +1,179 @@
+    AccessKeyId: String,
+    SecretAccessKey: String,
+    key: Match.OneOf(String, Function),
+    expire: Match.Where(function (expire) {
+      check(expire, Number);
+      return expire > 0;
+    }),
+  },
+
+  directiveDefault: {
+    region: 'auto',
+    expire: 5 * 60 * 1000,
+  },
+
+  getContentDisposition: async function (method, directive, file, meta) {
+    var getContentDisposition = directive.contentDisposition;
+
+    if (!_.isFunction(getContentDisposition)) {
+      getContentDisposition = function () {
+        var filename = file.name && encodeURIComponent(file.name);
+
+        return (
+          directive.contentDisposition ||
+          (filename &&
+            'inline; filename="' +
+              filename +
+              "\"; filename*=utf-8''" +
+              filename)
+        );
+      };
+    }
+
+    return getContentDisposition.call(method, file, meta);
+  },
+
+  upload: async function (method, directive, file, meta) {
+    try {
+      const policy = new Slingshot.StoragePolicy()
+        .expireIn(directive.expire)
+        .contentLength(0, Math.min(file.size, directive.maxSize || Infinity));
+
+      let key;
+      if (_.isFunction(directive.key)) {
+        key = await Promise.resolve(directive.key.call(method, file, meta));
+      } else {
+        key = directive.key;
+      }
+
+      const signedUrl = await this.getSignedUrl(directive, key, file.type);
+
+      const downloadUrl = `${directive.cdn}/${key}`
+
+      return {
+        upload: signedUrl,
+        download: downloadUrl,
+        postData: [],
+      };
+    } catch (error) {
+      console.error('Error in CloudflareR2 upload:', error);
+      throw error;
+    }
+  },
+
+  getSignedUrl: async function (directive, key, contentType) {
+    try {
+      const date = new Date();
+      const dateString = date.toISOString().replace(/[:-]|\.\d{3}/g, '');
+      const datestamp = dateString.slice(0, 8);
+      const credential = `${directive[this.accessId]}/${datestamp}/${
+        directive.region
+      }/s3/aws4_request`;
+
+      const queryParams = new url.URLSearchParams({
+        'X-Amz-Algorithm': 'AWS4-HMAC-SHA256',
+        'X-Amz-Credential': credential,
+        'X-Amz-Date': dateString,
+        'X-Amz-Expires': directive.expire.toString(),
+        'X-Amz-SignedHeaders': 'host',
+      });
+
+      const canonicalRequest = this.getCanonicalRequest(
+        directive,
+        key,
+        queryParams
+      );
+      const stringToSign = this.getStringToSign(
+        dateString,
+        datestamp,
+        directive.region,
+        canonicalRequest
+      );
+      const signature = this.getSignature(
+        directive[this.secretKey],
+        datestamp,
+        directive.region,
+        stringToSign
+      );
+
+      queryParams.set('X-Amz-Signature', signature);
+
+      return `${directive.endpoint}/${
+        directive.bucket
+      }/${key}?${queryParams.toString()}`;
+    } catch (error) {
```

Comment:
`try/catch` is not really useful here too

Reply:
fixed